src/server/routers/model-version-purchase.router.ts

```diff
@@ -0,0 +1,7 @@
+import { protectedProcedure, router } from '~/server/trpc';
+import { purchaseModelVersionInput } from '~/server/schema/model-version-purchase.schema';
+import { purchaseHandler } from '~/server/controllers/model-version-purchase.controller';
+
+export const modelVersionPurchaseRouter = router({
+  purchase: protectedProcedure.input(purchaseModelVersionInput).mutation(purchaseHandler),
```

Comment:
Let's put this behind a feature flag
```suggestion
  purchase: protectedProcedure.input(purchaseModelVersionInput).use(isFlagProtected('buzz')).mutation(purchaseHandler),
```